File: ingest/src/main/java/com/microsoft/azure/kusto/ingest/source/SourceInfo.java
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

package com.microsoft.azure.kusto.ingest.source;

import java.util.UUID;

public interface SourceInfo {
    /**
     * Checks that this SourceInfo is defined appropriately.
     */
    void validate();

    UUID getSourceId();

    void setSourceId(UUID sourceId);
}

File: ingest-v2/src/main/kotlin/com/microsoft/azure/kusto/ingest/v2/source/SourceInfo.kt
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

package com.microsoft.azure.kusto.ingest.v2.source

import java.util.UUID

interface SourceInfo {
    /**
     * Checks that this SourceInfo is defined appropriately.
     */
    fun validate()

    var sourceId: UUID?
}
